Flow.Launcher.Core/Plugin/PluginsLoader.cs

```diff
@@ -127,65 +110,74 @@ public static IEnumerable<PluginPair> DotNetPlugins(List<PluginMetadata> source)
-            string filename;
-
-            if (string.IsNullOrEmpty(pythonDirecotry))
+            // try to set Constant.PythonPath, either from
+            // PATH or from the given pythonDirectory
+            if (string.IsNullOrEmpty(pythonDirectory))
             {
                 var paths = Environment.GetEnvironmentVariable(PATH);
                 if (paths != null)
                 {
-                    var pythonPaths = paths.Split(';').Where(p => p.ToLower().Contains(Python));
-                    if (pythonPaths.Any())
+                    var pythonInPath = paths
+                        .Split(';')
+                        .Where(p => p.ToLower().Contains(Python))
+                        .Any();
+
+                    if (pythonInPath)
                     {
-                        filename = PythonExecutable;
+                        Constant.PythonPath = PythonExecutable;
                     }
                     else
                     {
                         Log.Error("|PluginsLoader.PythonPlugins|Python can't be found in PATH.");
-                        return new List<PluginPair>();
                     }
                 }
                 else
                 {
                     Log.Error("|PluginsLoader.PythonPlugins|PATH environment variable is not set.");
-                    return new List<PluginPair>();
                 }
             }
             else
             {
-                var path = Path.Combine(pythonDirecotry, PythonExecutable);
+                var path = Path.Combine(pythonDirectory, PythonExecutable);
                 if (File.Exists(path))
                 {
-                    filename = path;
+                    Constant.PythonPath = path;
                 }
                 else
                 {
-                    Log.Error("|PluginsLoader.PythonPlugins|Can't find python executable in <b ");
-                    return new List<PluginPair>();
+                    Log.Error($"|PluginsLoader.PythonPlugins|Can't find python executable in {path}");
                 }
             }
-            Constant.PythonPath = filename;
-            var plugins = metadatas.Select(metadata => new PluginPair
+
+            // if we have a path to the python executable,
+            // load every python plugin pair.
+            if (String.IsNullOrEmpty(Constant.PythonPath))
             {
-                Plugin = new PythonPlugin(filename),
-                Metadata = metadata
-            });
-            return plugins;
+                return new List<PluginPair>();
+            }
+            else
+            {
+                return source
+                    .Where(o => o.Language.ToUpper() == AllowedLanguage.Python)
+                    .Select(metadata => new PluginPair
+                    {
+                        Plugin = new PythonPlugin(Constant.PythonPath),
+                        Metadata = metadata
+                    });
+            }
```

Comment:
Probably just matter of personal preference. Would 
```
if (!String.IsNullOrEmpty(Constant.PythonPath))
     return true condition

return false condition
```
be more readable?

Reply:
would be the same to me, honestly. Should I invert it?

Reply:
yeah let's just leave it. i am not that fuss either way.